core/zoom_controls.js

```diff
@@ -453,7 +453,26 @@ Blockly.ZoomControls.prototype.createZoomResetSvg_ = function(rnd) {
  */
 Blockly.ZoomControls.prototype.resetZoom_ = function(e) {
   this.workspace_.markFocused();
-  this.workspace_.setScale(this.workspace_.options.zoomOptions.startScale);
+
+  // Update scrollX/scrollY so that the setScale translates the workspace
+  // correctly. This is needed for single-direction scrollbars.
+  var oldScale = this.workspace_.scale;
```

Comment:
Some questions:

- Why is this the responsibility of the zoom controls and not the workspace or the scrollbars?
  - Could it just be part of the scrollCenter function?
  - Are the zoom controls directly setting scrollX and scrollY anywhere else?
- Can you check the scrollbars and only do this when there are single-direction scrollbars?

Reply:
My issues stem from how `setScale` does a translation of the workspace that ends up moving the "center" of the workspace.
This wasn't a problem before because the scroll area used to be equivalent to the contentArea and scrolling the workspace wouldn't change the center.

In order to get around this problem, I need something like `zoomCenter` but where I pass in the exact scale.

I was initially intimidated by trying to compute the correct amount to pass to `zoomCenter`, so I went this route where I used logic from `zoom` to translate the workspace (by setting `scrollX` and `scrollY`) before calling `setScale` and I figured we may end up moving this logic into workspace.

HOWEVER, I tried my hand at computing the amount to pass to `zoomCenter` and with the extra comments I've added I think it's not too hard to follow.